### ISSUE TITLE
openssl:x86-windows-static build failure

### ISSUE THREAD

**Issue by gdiObjects** [microsoft/vcpkg#14451]
**Host Environment**
- OS: Windows 10 Pro
- Compiler: Microsoft Visual C++ 2019

**To Reproduce**
`.\vcpkg upgrade --no-dry-run

**Failure logs** 
D:\Development\Projects\vcpkg>.\vcpkg upgrade --no-dry-run
The following packages will be rebuilt:
  * libpq[core,openssl,zlib]:x86-windows-static
  * openexr[core]:x86-windows-static
  * openssl[core]:x86-windows-static
  * poco[core]:x86-windows-static
  * python3[core]:x86-windows-static
Additional packages (*) will be modified to complete this operation.
Starting package 1/10: poco:x86-windows-static
Removing package poco:x86-windows-static...
Removing package poco:x86-windows-static... done
Elapsed time for package poco:x86-windows-static: 1.092 s
Starting package 2/10: libpq:x86-windows-static
Removing package libpq:x86-windows-static...
Removing package libpq:x86-windows-static... done
Elapsed time for package libpq:x86-windows-static: 677.3 ms
Starting package 3/10: openexr:x86-windows-static
Removing package openexr:x86-windows-static...
Removing package openexr:x86-windows-static... done
Elapsed time for package openexr:x86-windows-static: 323.8 ms
Starting package 4/10: python3:x86-windows-static
Removing package python3:x86-windows-static...
Removing package python3:x86-windows-static... done
Elapsed time for package python3:x86-windows-static: 1.871 s
Starting package 5/10: openssl:x86-windows-static
Removing package openssl:x86-windows-static...
Removing package openssl:x86-windows-static... done
Elapsed time for package openssl:x86-windows-static: 18.96 ms
Detecting compiler hash for triplet x86-windows-static...
Starting package 6/10: openssl:x86-windows-static
Building package openssl[core]:x86-windows-static...
Could not locate cached archive: C:\Users\Rio\AppData\Local\vcpkg\archives\45\45f68f19a5f13a37467cc8fb47a4f2e25dcec86c.zip
-- Using community triplet x86-windows-static. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: D:\Development\Projects\vcpkg\triplets\community\x86-windows-static.cmake
CMake Error at ports/openssl/portfile.cmake:2 (message):
  Can't build openssl if libressl/boringssl is installed.  Please remove
  libressl/boringssl, and try install openssl again if you need it.
Call Stack (most recent call first):
  scripts/ports.cmake:135 (include)

Error: Building package openssl:x86-windows-static failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: openssl:x86-windows-static
  Vcpkg version: 2020.06.15-nohash

**Comment by sfhacker** [comment]
@gdiObjects Are we **STILL** having issues building OpenSSL from source on Windows?
Have you reported issues here before? Same port? Different ports?

**Comment by Neumann-A** [comment]
duplicate of #14440. Please remove `openssl-windows` via `vcpkg remove openssl-windows` and retry to install `openssl`

**Comment by gdiObjects** [comment]
Had issues building.
Yes, I have reported issues before.
Not with OpenSSL.

I followed Neumann-A's post:

D:\Development\Projects\vcpkg>.\vcpkg remove openssl-windows
The following packages are not installed, so not removed:
    openssl-windows:x86-windows
Another installed package matches the name of an unmatched request. Did you mean openssl-windows:x86-windows-static?
Package openssl-windows:x86-windows is not installed

D:\Development\Projects\vcpkg>.\vcpkg remove openssl-windows:x86-windows-static
The following packages will be removed:
    openssl-windows:x86-windows-static
Removing package openssl-windows:x86-windows-static...
Removing package openssl-windows:x86-windows-static... done

D:\Development\Projects\vcpkg>git pull
Already up to date.

D:\Development\Projects\vcpkg>.\vcpkg list
curl:x86-windows-static                            7.73.0#1         A library for transferring data with URLs
curl[non-http]:x86-windows-static                                   Enables protocols beyond HTTP/HTTPS/HTTP2
curl[schannel]:x86-windows-static                                   SSL support (Secure Channel)
curl[ssl]:x86-windows-static                                        Default SSL backend
curl[sspi]:x86-windows-static                                       SSPI support
curl[winssl]:x86-windows-static                                     Legacy name for schannel
curlpp:x86-windows-static                          2018-06-15-3     C++ wrapper around libcURL
exiv2:x86-windows-static                           0.27.3#2         Image metadata library and tools
exiv2[unicode]:x86-windows-static                                   Compile with unicode support on windows
exiv2[xmp]:x86-windows-static                                       Build with XMP metadata support
expat:x86-windows-static                           2020-08-18       XML parser library written in C
gettext:x86-windows-static                         0.19-15          The GNU gettext utilities are a set of tools tha...
lcms:x86-windows-static                            2.11             Little CMS.
libffi:x86-windows-static                          3.3#5            Portable, high level programming interface to va...
libiconv:x86-windows-static                        1.16#5           GNU Unicode text conversion
libpng:x86-windows-static                          1.6.37#12        libpng is a library implementing an interface fo...
pcre:x86-windows-static                            8.44#8           Perl Compatible Regular Expressions
restclient-cpp:x86-windows-static                  0.5.2            Simple REST client for C++. It wraps libcurl for...
sqlite3:x86-windows-static                         3.33.0           SQLite is a software library that implements a s...
tinyexr:x86-windows-static                         2020-03-15       Library to load and save OpenEXR(.exr) images
zlib:x86-windows-static                            1.2.11#9         A compression library

D:\Development\Projects\vcpkg>.\vcpkg install openssl[core]:x86-windows-static
Computing installation plan...
The following packages will be built and installed:
    openssl[core]:x86-windows-static
Detecting compiler hash for triplet x86-windows-static...
Starting package 1/1: openssl:x86-windows-static
Building package openssl[core]:x86-windows-static...
Could not locate cached archive: C:\Users\Rio\AppData\Local\vcpkg\archives\45\45f68f19a5f13a37467cc8fb47a4f2e25dcec86c.zip
-- Using community triplet x86-windows-static. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: D:\Development\Projects\vcpkg\triplets\community\x86-windows-static.cmake
-- Using cached D:/Development/Projects/vcpkg/downloads/openssl-1.1.1h.tar.gz
-- Extracting source D:/Development/Projects/vcpkg/downloads/openssl-1.1.1h.tar.gz
-- Using source at D:/Development/Projects/vcpkg/buildtrees/openssl/src/openssl-1-09fb822c56.clean
-- Copying openssl release source files...
-- Copying openssl release source files... done
-- Configure x86-windows-static-rel
-- Configure x86-windows-static-rel done
-- Build x86-windows-static-rel
-- Build x86-windows-static-rel done
-- Copying openssl debug source files...
-- Copying openssl debug source files... done
-- Configure x86-windows-static-dbg
-- Configure x86-windows-static-dbg done
-- Build x86-windows-static-dbg
-- Build x86-windows-static-dbg done
-- Installing: D:/Development/Projects/vcpkg/packages/openssl_x86-windows-static/share/openssl/copyright
-- Installing: D:/Development/Projects/vcpkg/packages/openssl_x86-windows-static/share/openssl//usage
-- Performing post-build validation
-- Performing post-build validation done
Stored binary cache: C:\Users\Rio\AppData\Local\vcpkg\archives\45\45f68f19a5f13a37467cc8fb47a4f2e25dcec86c.zip
Building package openssl[core]:x86-windows-static... done
Installing package openssl[core]:x86-windows-static...
Installing package openssl[core]:x86-windows-static... done
Elapsed time for package openssl:x86-windows-static: 2.905 min

Total elapsed time: 2.967 min

The package openssl is compatible with built-in CMake targets:

    find_package(OpenSSL REQUIRED)
    target_link_libraries(main PRIVATE OpenSSL::SSL OpenSSL::Crypto)
--
Then I went back and installed each missing package.  All is well.

-Thanks.
